### PR TITLE
Feat: add hex loader

### PR DESCRIPTION
**Pull request by LouisTsai-Csie**
# Summary
The original Ghidra extension does not support the hexadecimal file format, this PR add the functionality by examining the codebase.

# Introduction
The original Ghidrevm only support the raw binary file format, users should convert the raw binary file into hexadecimal format for further analysis. In this PR, the new `EVM loader` will dynamically analyze the file content to determine which of the file type it is and load the program with corresponding loader.

Users can download the deployed code on block explorer and start analyzing using Ghidrevm without any format conversion.

# Verification

I take Uniswap V2 Router on the Ethereum mainnet as our test case. 

Address: [0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D](https://etherscan.io/address/0x7a250d5630b4cf539739df2c5dacb4c659f2488d#code)

1. Download the deployed code in `uni.evm_h` file.
2. Create another file named `uni.evm`.
3. Run `cat uni.evm_h | xxd -p -r > uni.evm`.
4. Run the project as Ghidra.
5. Import `uni.evm_h` by `EVM loader`, and select the `EVM` language.
6. Import `uni.evm` by `EVM loader`, and select the `EVM` language.
7. Analyze these two files with `EVM Disassembler` and `EVM Function Analyzer`.
8. Compare the result.

The analysis result of `uni.evm_h`: 
Screenshot 2024-06-24 at 5 07 45 PM

The analysis result of `uni.evm`:
Screenshot 2024-06-24 at 5 08 43 PM

### CODE REVIEW

File: src/main/java/ghidrevm/GhidrevmLoader.java [hunk 1]
```diff
@@ -25,39 +27,88 @@
...
+
+					flatAPI.addEntryPoint(addr);
+				} else {
+					MemoryBlock block = flatAPI.createMemoryBlock("code", addr, data, false);
+					block.setRead(true);
+					block.setWrite(false);
+					block.setExecute(true);
+					
+					flatAPI.addEntryPoint(addr);
+				}
```
Comment (syjcnss) [inline src/main/java/ghidrevm/GhidrevmLoader.java]: 这段代码逻辑可以合并

File: src/main/java/ghidrevm/GhidrevmLoader.java [hunk 2]
```diff
@@ -25,39 +27,88 @@
...
 	}
 
 	@Override
 	public Collection<LoadSpec> findSupportedLoadSpecs(ByteProvider provider) throws IOException {
 		List<LoadSpec> loadSpecs = new ArrayList<>();
-
-		// TODO: Examine the bytes in 'provider' to determine if this loader can load it.  If it 
-		// can load it, return the appropriate load specifications.
-
+
```
Comment (syjcnss) [inline src/main/java/ghidrevm/GhidrevmLoader.java:53]: 在最开始判断一下provider.length()的大小，EVM bytecode应该是有个大小限制的，不然现在所有文件import都会出现EVM Loader

File: src/main/java/ghidrevm/GhidrevmLoader.java (hunk 1 above)
Comment (LouisTsai-Csie) [reply src/main/java/ghidrevm/GhidrevmLoader.java]: 我後續合併了，再麻煩確認一下，謝謝

File: src/main/java/ghidrevm/GhidrevmLoader.java (hunk 2 above)
Comment (LouisTsai-Csie) [reply src/main/java/ghidrevm/GhidrevmLoader.java:53]: 我後來依照 [EIP-170](https://eips.ethereum.org/EIPS/eip-170) 的內容判斷 `provider.length()`

合約大小的限制是 24576 bytes = 24576 * 2 hex = 24576 * 8 bits

**Pull request merged**